agent/handlers/agentapi/v1/taskprotection/handlers/handlers.go

```diff
@@ -59,53 +60,56 @@ func PutTaskProtectionHandler(state dockerstate.TaskEngineState,
 			return
 		}
 
-		if request.ProtectionType == "" {
+		if request.ProtectionEnabled == nil {
 			writeJSONResponse(w, http.StatusBadRequest,
-				"Invalid request: protection type is missing or empty",
+				"Invalid request: does not contain 'ProtectionEnabled' field",
 				putTaskProtectionRequestType)
 			return
 		}
 
-		taskProtection, err := types.NewTaskProtection(request.ProtectionType, request.ProtectionTimeoutMinutes)
+		taskProtection, err := types.NewTaskProtection(*request.ProtectionEnabled, request.ExpiresInMinutes)
 		if err != nil {
 			writeJSONResponse(w, http.StatusBadRequest,
 				fmt.Sprintf("Invalid request: %v", err),
 				putTaskProtectionRequestType)
 			return
 		}
 
-		task, err := getTaskFromRequest(state, r)
+		task, responseCode, err := getTaskFromRequest(state, r)
 		if err != nil {
-			writeJSONResponse(w, http.StatusInternalServerError,
-				fmt.Sprintf("Failed to find task: %v", err), putTaskProtectionRequestType)
+			writeJSONResponse(w, responseCode, err.Error(), putTaskProtectionRequestType)
 			return
 		}
 
 		// TODO: Call ECS
-		logger.Info("Would have called ECS.PutTaskProtection with fields", logger.Fields{
-			"cluster":                  cluster,
-			"serviceName":              task.ServiceName,
-			"taskId":                   task.Arn,
-			"protectionType":           taskProtection.GetProtectionType(),
-			"protectionTimeoutMinutes": taskProtection.GetProtectionTimeoutMinutes(),
+		logger.Info("PutTaskProtection endpoint was called", logger.Fields{
+			loggerfield.Cluster:        cluster,
+			loggerfield.TaskARN:        task.Arn,
+			loggerfield.TaskProtection: taskProtection,
 		})
 		writeJSONResponse(w, http.StatusOK, "Ok", putTaskProtectionRequestType)
 	}
 }
 
 // Helper function for finding task for the request
-func getTaskFromRequest(state dockerstate.TaskEngineState, r *http.Request) (*apitask.Task, error) {
+func getTaskFromRequest(state dockerstate.TaskEngineState, r *http.Request) (*apitask.Task, int, error) {
 	taskARN, err := v3.GetTaskARNByRequest(r, state)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get task ARN from request: %w", err)
+		logger.Error("Failed to find task ARN for task protection request", logger.Fields{
+			loggerfield.Error: err,
+		})
+		return nil, http.StatusBadRequest, errors.New("Invalid request: no task was found")
 	}
 
 	task, found := state.TaskByArn(taskARN)
 	if !found {
-		return nil, fmt.Errorf("could not find task from task ARN '%v'", taskARN)
+		logger.Error("No task was found for taskARN for task protection request", logger.Fields{
```

Comment:
Do you think this should be a critical? I feel like this is a big problem which essentially means our task engine state is corrupted.

Reply:
Yes if this happens then it's definitely a bug in our state management. I will change the log level to critical.

Reply:
Done.